chore: Ensure consistency across several references and update relevant SKILLs (#12082)
GoogleCloudPlatform/k8s-config-connector: .gemini/skills/kcc-direct-base-types-implementer/SKILL.md

  ```
  *(Note: See greenfield/brownfield skills for the correct `stability-level` label to append.)*
- **Status Fields**: `status.observedGeneration` must be exactly `*int64`.
- **Reference Fields**: Ensure that fields referencing other GCP/KCC resources are implemented as proper KCC reference fields (e.g., using `pubsubv1beta1.PubSubTopicRef` or `refsv1beta1.ProjectRef`), following the `Ref` suffix naming convention. You **MUST NOT** add new exceptions to `tests/apichecks/testdata/exceptions/missingrefs.txt`. All reference-like fields must be implemented as proper references.
- **Implement a New Reference Types**:

>   * Whenever a new reference type (e.g. `<Kind>Ref`) is needed, likely this reference resource does not have full KCC support yet. Only external reference is allowed.

Reviewer: ```suggestion
  * Whenever a new reference type (e.g. `<Kind>Ref`) is needed and the KCC reference resource does not have full KCC support yet, we can only support external reference.
```

>   The `<Kind>Ref` must **always** be defined and implemented in its own separate file named `<kind>_reference.go` (e.g., `computebackendbucket_reference.go`) under `apis/<service>/v1alpha1`(e.g., `apis/compute/v1alpha1/`) directory. This keeps the main type definitions clean and isolated from reference resolution boilerplate.

Reviewer 2: I'm okay with the v1alpha1 resource is supported this way (i.e. has the reference types it uses under v1alpha1), but we should also clarify in the promotion process that any reference types it uses must be graduated to v1beta1 as well.

Author: We don't have a SKILL for beta promotion yet, let me follow up on this in the promotion SKILL.
